### PR TITLE
feat: adds checks for owner role invite

### LINKED ISSUE #5890: Transfer the ownership of your events to another organizer.
**Is your feature request related to a problem? Please describe.**

We are implementing feature to provide ability to a user to delete his/her account. If the user is associated with any event, he/she must be given choice to transfer the events to other organizers associated with the event. (Just like in GitHub)

**Describe the solution you'd like**

Something similar to `Github's Danger Zone` 
danger

**Additional context**

Frontend issue: https://github.com/fossasia/open-event-frontend/issues/3012

### PR DESCRIPTION
**Pull request by shreyanshdwivedi**
Fixes #5890 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

#### Changes proposed in this pull request:
- Only owner of an event can send role-invite for owner to other user
- Ensures that when an owner role-invite is accepted, the previous owner is deleted so that there can be a single owner of an event. Also, previous owner role-invite and new owner's last role-invite is also deleted

### CODE REVIEW

**Comment by shreyanshdwivedi** [comment]
@iamareebjamal can you please add tag `DO_NOT_MERGE` till this PR - https://github.com/fossasia/open-event-server/pull/6042, is merged

**Comment by shreyanshdwivedi** [comment]
@iamareebjamal @uds5501 @mrsaicharan1 please review. This PR will add a check on accepting role-invite when role will be owner. It'll remove last owner UER and also last owner role-invite (if it exists)

File: app/api/helpers/role_invite.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        email=previous_uer.user.email, event_id=previous_uer.event_id, role_name='owner', status='accepted'
+        ).first()
+    if role_invite:
+        db.session.delete(role_invite)
+    db.session.delete(previous_uer)
+    try:
+        db.session.commit()
+    except Exception as e:
+        logging.error('DB Exception! %s' % e)
+        db.session.rollback()
```
Comment (mrsaicharan1) [inline app/api/helpers/role_invite.py]: You can use `save_to_db` here

File: app/api/role_invites.py [hunk 2]
```diff
@@ -165,6 +166,10 @@ def accept_invite():
             if not user.is_verified:
                 user.is_verified = True
                 save_to_db(user, 'User verified')
+            if role_invite.role_name == 'owner':
+                previous_uer = UsersEventsRoles.query.filter_by(event=event).filter_by(role=role).first()
```
Comment (mrsaicharan1) [inline app/api/role_invites.py]: Maybe use `filter_by(event=event, role=role)` here?

**Comment by shreyanshdwivedi** [comment]
@kushthedude @iamareebjamal @mrsaicharan1 please review this

**Comment by shreyanshdwivedi** [comment]
@iamareebjamal please review this

File: app/api/helpers/role_invite.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+        email=uer.user.email, event_id=uer.event_id, role_name='owner', status='accepted'
+        ).first()
+
+    if role_invite:
+        db.session.delete(role_invite)
+    db.session.delete(uer)
+    try:
+        db.session.commit()
+    except Exception as e:
+        logging.error('DB Exception! %s' % e)
```
Comment (iamareebjamal) [inline app/api/helpers/role_invite.py]: No root logging

**Comment by shreyanshdwivedi** [comment]
@iamareebjamal please have a look now. Removed root logging and updated the exception msg too

**Pull request merged**